src/main/java/org/wildfly/security/auth/client/AuthenticationConfiguration.java

```diff
@@ -1358,7 +1358,9 @@ SaslClient createSaslClient(URI uri, Collection<String> serverMechanisms, UnaryO
         }
         saslClientFactory = new LocalPrincipalSaslClientFactory(new FilterMechanismSaslClientFactory(saslClientFactory, filter));
         final SaslClientFactory finalSaslClientFactory = saslClientFactory;
-        saslClientFactory = doPrivileged((PrivilegedAction<PrivilegedSaslClientFactory>) () -> new PrivilegedSaslClientFactory(finalSaslClientFactory), capturedAccessContext);
+        final boolean captureAccessControlContext = Boolean.parseBoolean(System.getProperty("wildfly.elytron.capture.access.control.context", "true"));
```

Comment:
This system property needs to be read in a doPrivileged block if a security manager is present. In addition it should only be read once and stored in a static, not read per request.